crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -981,35 +995,76 @@ fn create_inventory_fragments_message(
     Ok(Message::new(&topic, ops_msg.to_string()))
 }
 
-async fn publish_restart_operation_status(
-    json_response: &str,
-) -> Result<Vec<Message>, CumulocityMapperError> {
-    let response = RestartOperationResponse::from_json(json_response)?;
-    let topic = C8yTopic::SmartRestResponse.to_topic()?;
-
-    match response.status() {
-        OperationStatus::Executing => {
-            let smartrest_set_operation = SmartRestSetOperationToExecuting::new(
-                CumulocitySupportedOperations::C8yRestartRequest,
-            )
-            .to_smartrest()?;
-
-            Ok(vec![Message::new(&topic, smartrest_set_operation)])
+impl CumulocityConverter {
```

Comment:
I just realised that you can have multiple plain impl blocks for the same struct, which are not for any trait implementations and this module already had 3 different impl blocks. Looks fine, but I just don't understand the benefits though. I'm assuming it is for better code organization to isolate different responsibilities into different impl blocks instead of one big fat impl block.

Reply:
Indeed, something that needs to be fixed. But in a PR doing only that.

Why are we here? Previously, there was an implementations of a trait used *only* here. There were also functions which have been promoted to methods. When I introduced [these changes](https://github.com/thin-edge/thin-edge.io/pull/2108), I decided to go that way in order to help the code reviewers by minimizing code shuffling on an already huge refactoring.

Now, this can be cleaned up.